[WFLY-14673] Utilize org.wildfly.common.Assert for Null-Checks (clust…

boris-unckel opened the pull request:
…ering)

Fixing https://issues.redhat.com/browse/WFLY-14673

The PR 14180 was too big to verify. It has been splitted up, here: clustering.

boris-unckel committed:
[WFLY-14673] Utilize org.wildfly.common.Assert for Null-Checks (clustering)

Fixing https://issues.redhat.com/browse/WFLY-14673

bstansberry commented:
@pferraro Please review

pferraro commented on testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/AbstractClusteringTestCase.java:
```diff
@@ -289,10 +289,7 @@ public void stop(String... nodes) {
-                if (deployment == null) {
-                    throw new IllegalArgumentException(node);
-                }
+                String deployment = checkNotNullParam(String.valueOf(node), NODE_TO_DEPLOYMENT.get(node));
```
This is not a proper use case of Assert. "deployment" is not a parameter.

boris-unckel commented:
@pferraro Thanks for taking the time to review the PR. I agree that the variables you mentioned are not parameters. I have reverted the changes and updated the PR. I still think it was a valid approach: [Google Guavas Preconditions](https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Preconditions.java#L894) and [Apache Commons Lang Validate](https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/Validate.java) share the same idea. Aside from a dependency discussion they do not face the project reality where both IAE and NPE is in use. o.w.common.Assert perfectly adresses that, but is focussed on "param". I like the fluent API style of the check and the enforcement of an exception text.
Every reverted exception in this PR already had a meaningful text, so there is no loss for the concrete case.

pferraro commented on clustering/ee/cache/src/main/java/org/wildfly/clustering/ee/cache/scheduler/PortableConcurrentDirectDeque.java:
```diff
@@ -1124,7 +1116,7 @@ public boolean addAll(Collection<? extends E> c) {
         Node<E> beginningOfTheEnd = null, last = null;
         for (E e : c) {
-            checkNotNull(e);
+            checkNotNullParamWithNullPointerException("e", e);
```
"e" is not a parameter of this method.  Is there an Assert.checkNotNullArrayParam(...) equivalent for collections?

boris-unckel replied:
There is no equivalent method, but even when it would not be applicable to a non-parameter variable here. I have reverted the change.

pferraro commented on clustering/ee/cache/src/main/java/org/wildfly/clustering/ee/cache/scheduler/FastConcurrentDirectDeque.java:
```diff
@@ -1157,7 +1149,7 @@ public boolean addAll(Collection<? extends E> c) {
         Node<E> beginningOfTheEnd = null, last = null;
         for (E e : c) {
-            checkNotNull(e);
+            checkNotNullParamWithNullPointerException("e", e);
```
"e" is not a parameter of this method. Is there an Assert.checkNotNullArrayParam(...) equivalent for collections?

pferraro commented on clustering/ee/cache/src/main/java/org/wildfly/clustering/ee/cache/scheduler/FastConcurrentDirectDeque.java:
```diff
@@ -853,7 +845,7 @@ public FastConcurrentDirectDeque(Collection<? extends E> c) {
         Node<E> h = null, t = null;
         for (E e : c) {
-            checkNotNull(e);
+            checkNotNullParamWithNullPointerException("e", e);
```
"e" is not a parameter of this method. Is there an Assert.checkNotNullArrayParam(...) equivalent for collections?

pferraro commented on clustering/ee/cache/src/main/java/org/wildfly/clustering/ee/cache/scheduler/PortableConcurrentDirectDeque.java:
```diff
@@ -820,7 +812,7 @@ public PortableConcurrentDirectDeque(Collection<? extends E> c) {
         Node<E> h = null, t = null;
         for (E e : c) {
-            checkNotNull(e);
+            checkNotNullParamWithNullPointerException("e", e);
```
"e" is not a parameter of this method. Is there an Assert.checkNotNullArrayParam(...) equivalent for collections?

pferraro commented on ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/clustering/infinispan/spi/main/module.xml:
```diff
@@ -46,5 +46,6 @@
         <module name="org.wildfly.clustering.marshalling.spi"/>
         <module name="org.wildfly.clustering.service"/>
         <module name="org.wildfly.security.elytron-private"/>
+        <module name="org.wildfly.common"/>
```
Nit: Please retain the lexicographical ordering.

pferraro replied:
I'll fix this later.

boris-unckel replied:
@pferraro Sorry, I did not recognize you meant the existing entries also. Next time better. Thanks for taking the time to review and merge.

pferraro commented on clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/ProtocolMetricsHandler.java:
```diff
@@ -234,7 +234,6 @@ public ProtocolMetricsHandler(FunctionExecutorRegistry<JChannel> executors) {
 
     @Override
     protected void executeRuntimeStep(OperationContext context, ModelNode operation) throws OperationFailedException {
-
```
Nit: I assume this was unintentional?
